### DISCUSSION TITLE
Image Magick speed compared to Irfanview when combining tif files

### DISCUSSION THREAD

**Discussion by Jason-Orman** [ImageMagick/ImageMagick#6185] [Help]
### ImageMagick version

7.1.0-48 Q16-HDRI x64 c243c92:20220911

### Operating system, version and so on

Windows 11

### Description

I'm testing various ways to combine multiple tifs into a multipage file. 

With Image Magick, running a single process, one after another I can do roughly 40 images a minute. I am essentially running this in a loop within a python script:

_magick -define tiff:ignore-tags=34686  -define tiff:preserve-compression=true c:\input\'+fileName+'*.tif  c:\output\'+fileName+'.tif_

IrfanView does about 350 a minute using the similar functionality 
_i_view32.exe /multitif=(c:\output\'+filename+'.tif,c:\input\'+filename+'_*.*) /killmesoftly /silent

Is there perhaps something I am missing in how to speed up image magick I would prefer to use it because there are so many other processes we use it for?

(Ultimately my plan is to parallelize both of these spinning up a number of threads equal to CPU cores and run 20 or so of each of these in parallel, so speed is of the essence).

**Comment by snibgo** [comment]
ImageMagick will read all your input images into memory before writing them. If that exceeds available RAM, IM will use disk instead which will be slower.

For such a simple operation, using IM is overkill. Perhaps tiffcp will be faster than IM or IrfanView.

**Comment by Jason-Orman** [comment]
In this case, I feed one image at a time into Image Magick (or one set of
images a1*.tif for example)
